genty/genty.py

```diff
@@ -129,7 +129,7 @@ def _expand_repeats(test_functions):
                     dataprovider,
                     repeat_suffix,
                 )
-        elif dataset:
+        elif dataset or dataprovider:
```

Comment:
I was looking at this earlier, and was wondering: why does this need to be ``elif`` at all? Why not ``else``? Shouldn't earlier steps have been responsible for weeding out things that don't belong?